cmd/internal/pluginutils/errors.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginutils
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"k8s.io/klog/v2"
+)
+
+// GpuFatalErrors returns (value, name) of first GPU fatal_* error counter
+// file with non-zero value, or (0, "") if there were no fatal errors.
+func GpuFatalErrors(syspath string) (int64, string) {
+	var errname string
+	errors := int64(0)
+	for tile := 0; errors == 0; tile++ {
+		path := path.Join(syspath, fmt.Sprintf("gt/gt%d", tile))
+		errors, errname = tileFatalErrors(path)
+		if errors > 0 {
+			return errors, errname
+		}
+	}
```

Comment:
Should this check if "path" exists and break if it doesn't?

Reply:
It does.  `tileFatalErrors()` returns `-1` if there were no Glob matches, and loop runs only until return value is non-zero.  Do you have a suggestion how the code would be clearer?

Reply:
I see. Now that you mention it, _errors_ is used in the for loop condition. I got confused because the loop has one if for the positive values but nothing for the negative.

> Do you have a suggestion how the code would be clearer?

A comment would make it clearer. I might personally use a boolean variable in the for condition (pathFound) that I would update at the end of the for loop (pathFound = errors == 0).

Reply:
Added separate bool for whether the loop should proceed to checking next tile.